src/LazyJSON.jl

```diff
@@ -0,0 +1,324 @@
+        else
+            return nothing
+        end
+    end
+end
+
+Base.length(x::LazyJSONDict) = yyjson_obj_size(x.ptr)
+
+#__ LazyJSONVector
+
+"""
+    LazyJSONVector <: AbstractVector{Any}
+
+Represents a vector for a JSON array.
+
+## Fields
+- `ptr::Ptr{YYJSONVal}`: The array value pointer.
+- `doc_ptr::Ptr{YYJSONDoc}`: The JSON document pointer (non-null only for the root).
+- `alc_ptr::Ptr{YYJSONAlc}`: The dynamic allocator pointer (non-null only for the root).
+- `freed::Bool`: Flag indicating whether the document is freed.
+"""
+mutable struct LazyJSONVector <: AbstractVector{Any}
+    ptr::Ptr{YYJSONVal}
+    doc_ptr::Ptr{YYJSONDoc}
+    alc_ptr::Ptr{YYJSONAlc}
+    freed::Bool
+
+    function LazyJSONVector(ptr::Ptr{YYJSONVal})
+        new(ptr, YYJSONDoc_NULL, YYJSONAlc_NULL, false)
+    end
+
+    function LazyJSONVector(ptr::Ptr{YYJSONVal}, doc_ptr::Ptr{YYJSONDoc}, alc_ptr::Ptr{YYJSONAlc})
+        new(ptr, doc_ptr, alc_ptr, false)
+    end
+end
+
+function Base.close(arr::LazyJSONVector)
+    arr.freed && return nothing
+    yyjson_doc_free(arr.doc_ptr)
+    yyjson_alc_dyn_free(arr.alc_ptr)
+    arr.freed = true
+    return nothing
+end
+
+Base.length(x::LazyJSONVector) = yyjson_arr_size(x.ptr)
+
+Base.size(x::LazyJSONVector) = (yyjson_arr_size(x.ptr),)
+
+function Base.getindex(arr::LazyJSONVector, index::Integer)
+    value = get(arr, index, YYJSONVal_NULL)
+    value === YYJSONVal_NULL && throw(BoundsError(arr, index))
+    return value
+end
+
+function Base.get(arr::LazyJSONVector, index::Integer, default)
+    (1 <= index <= length(arr)) || return default
+    value_ptr = yyjson_arr_get(arr.ptr, index-1)
+    return value_ptr !== YYJSONVal_NULL ? parse_json_value(value_ptr) : default
+end
+
+#__ API
+
+function parse_json_value(ptr::Ptr{YYJSONVal})
+    return if yyjson_is_str(ptr)
+        parse_json_string(ptr)
+    elseif yyjson_is_raw(ptr)
+        parse_json_string(ptr)
+    elseif yyjson_is_num(ptr)
+        parse_json_number(ptr)
+    elseif yyjson_is_bool(ptr)
+        yyjson_get_bool(ptr)
+    elseif yyjson_is_obj(ptr)
+        LazyJSONDict(ptr)
+    elseif yyjson_is_arr(ptr)
+        LazyJSONVector(ptr)
+    else
+        nothing
+    end
+end
+
+function parse_json_string(ptr::Ptr{YYJSONVal})
+    ptr_char = yyjson_get_str(ptr)
+    ptr_char === YYJSONUInt8_NULL && throw(LazyJSONError("Error parsing string."))
+    return unsafe_string(ptr_char)
+end
+
+function parse_json_number(ptr::Ptr{YYJSONVal})
+    return yyjson_is_real(ptr) ? yyjson_get_real(ptr) : yyjson_get_int(ptr)
+end
+
+function parse_json_root(doc_ptr::Ptr{YYJSONDoc})
+    root_ptr = yyjson_doc_get_root(doc_ptr)
+    root_ptr === YYJSONVal_NULL && throw(LazyJSONError("Error parsing root."))
+    return root_ptr
+end
+
+"""
+    parse_lazy_json(json::AbstractString; kw...)
+    parse_lazy_json(json::AbstractVector{UInt8}; kw...)
+
+Parse a JSON string `json` (or vector of `UInt8`) into a [`LazyJSONDict`](@ref) or [`LazyJSONVector`](@ref).
+
```

Comment:
Add a warning about the need to close the object for reading after finishing work
And a reference to the method that takes a function as the first argument, which itself will close the lazy object